berkeley-mobile/StudyPact/StudyPact.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  StudyPact.swift
+//  berkeley-mobile
+//
+//  Created by Shawn Huang on 2/2/21.
+//  Copyright © 2021 ASUC OCTO. All rights reserved.
+//
+
+import Foundation
+import GoogleSignIn
+
+/// All StudyPact api call functions go in here. Relevant user info also goes here.
+class StudyPact {
+    static let shared = StudyPact()
+    
+    private var cryptoHash: String?
+    var email: String?
+    
+    /// Load cryptohash from user defaults. If it doesnt exist or authenticate fails, return false.
+    public func loadCryptoHash() -> Bool {
+        reset()
+        if let cryptoHash = UserDefaults.standard.string(forKey: "userCryptoHash") {
```

Comment:
`reset()` deletes the CryptoHash, so this if statement is never true and the user must re-authenticate each time the app is started.

Reply:
oh yeah lmao that was for testing